lib/CodeGen/CGDynamicCheck.cpp

```diff
@@ -54,6 +54,9 @@ void CodeGenFunction::EmitDynamicNonNullCheck(const Address BaseAddr, const Qual
   if (!getLangOpts().CheckedC)
     return;
 
+  if (!CGM.getCodeGenOpts().CheckedCRuntimeChecks)
```

Comment:
Runtime checks for non-null will only be added if the flag `-fcheckedc-runtime-checks` is present. Since I have added this check here, it means this would affect all non-null checks (and not just those for pointer arithmetic). This is resulting in 3 unit test failures which we need to fix (by adding this flag to those tests). Is this behavior acceptable?

Reply:
It should be on by default, and something that we can disable.